Simplify assets loading

jurgenwerk opened the pull request:
This PR makes ember app assets loading much simpler: by providing the final asset url in the assets links, instead of linking to the realm. Previously, the host app issued asset requests to the realm, and the realm redirected the request to where the assets are actually hosted (on S3 for example, in our deployed environments).

This PR comes with the following advantages:

1. The simplification allows us to remove our app asset redirections middleware (less server logic, less overhead)
2. Also removes the need for monaco editor asset proxying (we were discussing that we have this because monaco editor's worker needs to have its assets loaded via same origin, but this is not needed anymore)

To illustrate, let's take a look at the app's html source on staging (when this branch is deployed):

<img width="856" alt="image" src="https://github.com/cardstack/boxel/assets/273660/da20a842-f9ba-4f79-84c9-dfde3354c50d">

<img width="1051" alt="image" src="https://github.com/cardstack/boxel/assets/273660/0d07a803-66af-4f16-8db0-4b39013a92fa">

## Monaco code editor after removing asset proxying middleware

<img width="1309" alt="image" src="https://github.com/cardstack/boxel/assets/273660/1faff2be-db32-43c9-946b-b38a3ca5d192">

jurgenwerk committed:
wip

jurgenwerk committed:
Merge branch 'main' into assets-refactor

jurgenwerk committed:
Simplify and cleanup

jurgenwerk committed:
Merge branch 'main' into assets-refactor-1

jurgenwerk committed:
Not needed

jurgenwerk committed:
Provide assetsURL in tests

jurgenwerk committed:
Adjust test after assets url adjustment

jurgenwerk committed:
Unnecessary escape character

tintinthong reviewed (approved):
Super! Thanks for the great description in this PR

jurgenwerk commented on packages/realm-server/main.ts:
```diff
@@ -155,6 +157,20 @@ if (distURL) {
+if (distURL) {
+  assetsURL = new URL(distURL);
+} else {
+  // Default to the base dist URL for assets
```
This will be the case in development. app loaded from base/drafts/published realms should consume assets from base realm

backspace reviewed (approved):
nice simplification, and agreed re the thorough explanation 🤩

jurgenwerk committed:
Load ttf inline (codicon font for monaco) to prevent asset

In development, it tried to load the font from localhost:4201 instead of the assets url (localhost:4201/base/__boxel) even though __webpack_public_path__ is set to the assets url, producing a 404. We circumvent this problem by loading it iline with data:font

jurgenwerk committed:
Merge branch 'main' into assets-refactor-1
